test/governance/extensions/GovernorCountingFractional.test.js

```diff
@@ -0,0 +1,195 @@
+      beforeEach(async function () {
+        Object.assign(this, await loadFixture(fixture));
+
+        // default proposal
+        this.proposal = this.helper.setProposal(
+          [
+            {
+              target: this.receiver.target,
+              value,
+              data: this.receiver.interface.encodeFunctionData('mockFunction'),
+            },
+          ],
+          '<proposal description>',
+        );
+      });
+
+      it('deployment check', async function () {
+        expect(await this.mock.name()).to.equal(name);
+        expect(await this.mock.token()).to.equal(this.token);
+        expect(await this.mock.votingDelay()).to.equal(votingDelay);
+        expect(await this.mock.votingPeriod()).to.equal(votingPeriod);
+      });
+
+      it('nominal is unaffected', async function () {
+        await this.helper.connect(this.proposer).propose();
+        await this.helper.waitForSnapshot();
+        await this.helper.connect(this.voter1).vote({ support: VoteType.For, reason: 'This is nice' });
+        await this.helper.connect(this.voter2).vote({ support: VoteType.For });
+        await this.helper.connect(this.voter3).vote({ support: VoteType.Against });
+        await this.helper.connect(this.voter4).vote({ support: VoteType.Abstain });
+        await this.helper.waitForDeadline();
+        await this.helper.execute();
+
+        expect(await this.mock.hasVoted(this.proposal.id, this.owner)).to.be.false;
+        expect(await this.mock.hasVoted(this.proposal.id, this.voter1)).to.be.true;
+        expect(await this.mock.hasVoted(this.proposal.id, this.voter2)).to.be.true;
+        expect(await ethers.provider.getBalance(this.mock)).to.equal(0n);
+        expect(await ethers.provider.getBalance(this.receiver)).to.equal(value);
+      });
+
+      it('voting with a fraction of the weight: twice', async function () {
+        await this.helper.connect(this.proposer).propose();
+        await this.helper.waitForSnapshot();
+
+        expect(await this.mock.proposalVotes(this.proposal.id)).to.deep.equal([0n, 0n, 0n]);
+        expect(await this.mock.hasVoted(this.proposal.id, this.voter2)).to.equal(false);
+        expect(await this.mock.voteWeightCast(this.proposal.id, this.voter2)).to.equal(0n);
+
+        const steps = [['0', '2', '1'].map(ethers.parseEther), ['1', '0', '1'].map(ethers.parseEther)];
+
+        for (const votes of steps) {
+          const params = ethers.solidityPacked(['uint128', 'uint128', 'uint128'], votes);
+          await expect(
+            this.helper.connect(this.voter2).vote({
+              support: 0,
+              reason: 'no particular reason',
+              params,
+            }),
+          )
+            .to.emit(this.mock, 'VoteCastWithParams')
+            .withArgs(
+              this.voter2,
+              this.proposal.id,
+              0,
+              ethers.parseEther('7'), // total weight for voter2
+              'no particular reason',
+              params,
```

Comment:
Events are not great. Indexing is likelly going to be broken :/

Reply:
Asked Tally how they are handling it.

Reply:
Answer: 
- if params is not empty
  - disregard `support` and `weight` from the event
  - decode the params as 3 uint128
  - use these values
- if params is empty
  - use support and weight from the event

This confirms that we should fix the event to emit the correct weight when doing a "norminal" vote with the remaining power.

One question remain: How do we deal with support if params are not empty. 
- Is it ok to pass the value sent by the user without checking it.
- Should we add a 4th value (Against, For, Abstain, Fractional), and require that if params is not empty, then support must be Fractional ? This would possibly break compatibility existing modules for fractional counting that do not support that 4th option.

Reply:
>This would possibly break compatibility existing modules for fractional counting that do not support that 4th option.

It would be different behavior, but we can specify that support should be ignored and in that sense it would remain compatible.

Reply:
I was afraid that Scopelift's version would not accept any support that is not 0, 1 or 2. But that is only true if there is no params. If they have params, then the support is discarded (just like we do now).

So right now, where there are params, both version (scopelift and this PR) will accept any support. The question is do we want to force the support to a specific value (which scopelift doesn't do) so that event are "clearer" ?

Reply:
In my opinion we should make clear what the event represents. Given that the optional parameters are arbitrary bytes we can't specify a format, so the general guideline here should to be look for making the optional parameters as clear as possible.

I'd be in favor of changing the format, we're not restricted yet by backwards compatibility and the only constraint would be that Tally (or others) should update to support this variant. I think that's better than us starting with a design we're not comfortable with.

What do you think about restricting nominal voting to only when no previous vote was casted? That'd make it consistent, either it's a single nominal vote or multiple fractional ones.

Reply:
Added "Fractional" to the VoteType enum following our discussion. WDYT?

Reply:
Addition looks good to me. Left a few comments but I'd say we can close this.